src/components/common/MapDateSelect/index.tsx

```diff
@@ -0,0 +1,88 @@
+import DateSlider from './DateSlider';
+import strings from 'src/strings';
+import { Box, Typography, useTheme } from '@mui/material';
+import { useMemo, useState } from 'react';
+
+type MapDateSelectProps = {
+  dates: Date[];
+  onChange: (newDate: Date) => void;
+};
+
+export default function MapDateSelect({ dates, onChange }: MapDateSelectProps): JSX.Element {
+  const theme = useTheme();
+
+  const monthStrings = [
+    strings.MONTH_01,
+    strings.MONTH_02,
+    strings.MONTH_03,
+    strings.MONTH_04,
+    strings.MONTH_05,
+    strings.MONTH_06,
+    strings.MONTH_07,
+    strings.MONTH_08,
+    strings.MONTH_09,
+    strings.MONTH_10,
+    strings.MONTH_11,
+    strings.MONTH_12,
+  ];
+
+  const earliestDate = useMemo(() => {
+    return dates.reduce((prev, curr) => {
+      if (prev.getTime() < curr.getTime()) {
+        return prev;
+      }
+      return curr;
+    });
+  }, [dates]);
+
+  const latestDate = useMemo(() => {
+    return dates.reduce((prev, curr) => {
```

Comment:
`Math` has a min and max function which supports array spread operator, you can optionally do:
```
  return Math.max([... dates.map(date => date.getTime())])
```

Reply:
Nice, thanks!

Reply:
Nice, thanks!

Reply:
Doesn't work in this case, since I want to return the `Date`, and `Math.max/min` will return the timestamp.

Reply:
That makes sense. Missed the date as result.